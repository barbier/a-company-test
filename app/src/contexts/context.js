import { createContext } from 'react'

export const ArticleContext = createContext()
