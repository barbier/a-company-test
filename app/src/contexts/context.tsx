import React, { useState, createContext, useContext } from "react";
import { IArticle } from "../interfaces/Article";
import { IContext } from "../interfaces/ContextProvider";

const useArticle = (): any => {
  const [selectedArticle, setSelectedArticle] = useState({});

  return {
    selectedArticle,
    chooseArticle: (article: IArticle) => setSelectedArticle(article),
  };
};

const ArticleContext = createContext({} as any);

export const ArticleContextProvider: React.FC<{ children: any }> = ({
  children,
}: IContext) => (
  <ArticleContext.Provider value={useArticle()}>
    {children}
  </ArticleContext.Provider>
);

export const useSelectedArticle = (): any =>
  useContext(ArticleContext).selectedArticle;
export const useSetSelectedArticle = (article: IArticle): any =>
  useContext(ArticleContext).chooseArticle(article);
