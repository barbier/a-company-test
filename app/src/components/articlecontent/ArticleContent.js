import { useContext } from 'react'
import { ArticleContext } from '../../contexts/context'
import { Content, ContentAuthor, ContentHeader, ContentTitle, ContentPublishedDate, ContentImage, ContentText, ContentFooter, ContentSource } from '../../styles/styles'
import { convertToDate } from '../../utils/util'

function ArticleContent() {
  // eslint-disable-next-line
  const [selectedArticle, setSelectedArticle] = useContext(ArticleContext)

  const {
    title,
    author,
    publishedAt,
    urlToImage,
    content,
    url,
    source,
  } = selectedArticle
  
  if (!selectedArticle.title) {
    return null
  }

  return (
    <Content>
      <ContentHeader>
        <ContentTitle>{title}</ContentTitle>
        <ContentAuthor>
          Author: 
          {
            author
            ? author
            : 'Unknown'
          }
        </ContentAuthor>
        <ContentPublishedDate>
          Published at: {convertToDate(publishedAt)}
        </ContentPublishedDate>
      </ContentHeader>
      <ContentImage src={urlToImage} />
      <ContentText>{content}</ContentText>
      <ContentFooter>
        Source: 
        <ContentSource href={url} target='_blank'>
          {source?.name}
        </ContentSource>
      </ContentFooter>
    </Content>
  )
}

export default ArticleContent