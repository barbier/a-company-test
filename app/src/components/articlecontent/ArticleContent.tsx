/* eslint-disable */
import React from "react";
import { convertToDate } from "../../utils/util";

import { useSelectedArticle } from "../../contexts/context";

const {
  Content,
  ContentAuthor,
  ContentHeader,
  ContentTitle,
  ContentPublishedDate,
  ContentImage,
  ContentText,
  ContentFooter,
  ContentSource,
} = require("../../styles/styles");
/* eslint-enable */

const ArticleContent: React.FC = () => {
  const selectedArticle = useSelectedArticle();
  const { title, author, publishedAt, urlToImage, content, url, source } =
    selectedArticle;

  return (
    <Content>
      <ContentHeader>
        <ContentTitle>{title}</ContentTitle>
        <ContentAuthor> Author: {author ?? "Unknown"} </ContentAuthor>
        <ContentPublishedDate>
          Published at: {convertToDate(publishedAt)}
        </ContentPublishedDate>
      </ContentHeader>
      <ContentImage src={urlToImage} />
      <ContentText>{content}</ContentText>
      <ContentFooter>
        Source:
        <ContentSource href={url} target="_blank">
          {source?.name}
        </ContentSource>
      </ContentFooter>
    </Content>
  );
};

export default ArticleContent;
