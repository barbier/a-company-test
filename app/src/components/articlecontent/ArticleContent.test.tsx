import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ArticleContent from './ArticleContent'

import { ArticleContextProvider } from "../../contexts/context";
describe("Test Article Content rendering options", () => {
  test("It should match snapshot", () => {
    const component = renderer.create(
      <ArticleContextProvider>
        <ArticleContent />
      </ArticleContextProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  // test("It shouldn't render nothing if there's no article selected", () => {
  //   const {container} = render(
  //     <ArticleContextProvider>
  //       <ArticleContent />
  //     </ArticleContextProvider>
  //   )
  //   const articleTag = container.querySelector('article')

  //   expect(articleTag).toBeFalsy()
  // })

  test("It shouldn't render normaly when an article is selected", () => {

    const {container} = render(
      <ArticleContextProvider>
        <ArticleContent />
      </ArticleContextProvider>
    )
    const articleTag = container.querySelector('article')

    expect(articleTag).toBeTruthy()
  })
})
