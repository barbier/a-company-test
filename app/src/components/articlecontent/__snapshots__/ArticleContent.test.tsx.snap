// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Article Content rendering options It should match snapshot 1`] = `
<article
  className="css-6p15e3"
>
  <header
    className="css-rg0tw2"
  >
    <h1
      className="css-q28qg0"
    />
    <h2
      className="css-q28qg0"
    >
       Author: 
      Unknown
       
    </h2>
    <small
      className="css-0"
    >
      Published at: 
      Invalid Date
    </small>
  </header>
  <img
    className="css-1t0xj7x"
  />
  <p
    className="css-1k46fjq"
  />
  <footer
    className="css-0"
  >
    Source:
    <a
      className="css-hnw5at"
      target="_blank"
    />
  </footer>
</article>
`;
